### DISCUSSION TITLE
How can i wait consumer to deliver the message with same key

### DISCUSSION THREAD

**Discussion by ngoyal16** [apache/pulsar#18538] [Q&A]
How to wait consumer to deliver the message with same key until there is an ACK or NCK for already consumed message with same key is came?

**Comment by ngoyal16** [comment]
can someone please help me with this? @Jason918

**Reply by Jason918** [reply]
Seems like you are talking about the consumption order guarantee.
If you are using a partitioned topic and did not change the producer's message routing mode, ' RoundRobinPartition` by default, make sure messages with the same key go into the same partition.
On the consumer end, as long as you don't use the `SubscriptionType.Shared`, the consumption order should be guaranteed.
